### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by echaulk2**
Refactored game methods and added unit tests.

### CODE REVIEW

File: functions/index.ts [hunk 1]
```diff
@@ -1,168 +1,105 @@
...
-        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
-      } else {
-        throw Error;
-     }
-    } catch (err) {
-      throw HttpResponse({statusCode: 404, body: `Unable to find game.`});
     }
+}  
+
+export async function GetGameHttpResponse(game: Game) {
```
Comment (leavertj) [inline functions/index.ts:31]: camelCase

File: functions/index.ts [hunk 2]
```diff
@@ -1,168 +1,105 @@
...
+  let response = await GetGame(game);
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Item) {
+    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to get game."});
```
Comment (leavertj) [inline functions/index.ts:38]: Are you sure this case always represents a missing game? (I would explicitly test for an empty struct here)

File: functions/game.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+       this.gameName = gameName,
+       this.yearReleased = yearReleased,
+       this.genre = genre,
+       this.console = console,
+       this.developer = developer
+    }
+    
+    //Database CRUD Methods
+    
+    async CreateGame() {
```
Comment (leavertj) [inline functions/game.ts:33]: camelCase

File: functions/index.ts [hunk 4]
```diff
@@ -1,168 +1,105 @@
...
+}
+
+export async function ListGamesHttpResponse() {
+  let response = await ListGames();
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Items) {
+    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to get list of games."})
```
Comment (leavertj) [inline functions/index.ts:49]: There is no issue with returning an empty list. No need for this case.

File: functions/index.ts [hunk 5]
```diff
@@ -1,168 +1,105 @@
...
-      ConditionExpression: 'attribute_not_exists(gameName)'
-    }
+export async function CreateGameHttpResponse(game: Game) {
+  let response = await game.CreateGame();
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Item) {
+    return HttpResponse({statusCode: 201, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to create game."});
```
Comment (leavertj) [inline functions/index.ts:60]: 404 means 'Not Found' which I don't think is appropriate here. What would be the case where this block is executed? Should it just be 500?

File: functions/index.ts [hunk 6]
```diff
@@ -1,168 +1,105 @@
...
-      throw HttpResponse({statusCode: 403, body: `Unable to modify game.`});
-    }
+export async function DeleteGameHttpResponse(game: Game) {
+  let response = await DeleteGame(game);
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Attributes) {
+    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to delete game."});
```
Comment (leavertj) [inline functions/index.ts:80]: Same question as on line 60.

File: test/functions.test.ts [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+            gameName: 'League of Legends',
+            yearReleased: '2008', 
+            genre: 'Moba', 
+            console: 'PC', 
+            developer: 'Riot Games'
+        }
+    })
+});
+
+test("GetGame", async () => {
```
Comment (leavertj) [inline test/functions.test.ts:18]: Could test the 404 path here.

File: test/functions.test.ts [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+            developer: 'Riot Games',
+            gameName: 'League of Legends',
+            yearReleased: '2008', 
+        }],
+        Count: 1,
+        ScannedCount: 1
+    })
+});
+
+test("ModifyGame", async () => {
```
Comment (leavertj) [inline test/functions.test.ts:47]: Test the case where you change the primary key

File: test/functions.test.ts [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+            gameName: 'League of Legends',
+            yearReleased: '2021', 
+            genre: 'Strategy', 
+            console: 'PC', 
+            developer: 'Riot Games'
+        }
+    })
+});
+
+test("DeleteGame", async () => {
```
Comment (leavertj) [inline test/functions.test.ts:65]: Test deleting a game that doesn't exist?

**Review by leavertj — APPROVED** [review]
This is awesome progress. I definitely think we can start expanding on the functionality here to support multiple users. (Remember to cleanup compiled JS scripts)
